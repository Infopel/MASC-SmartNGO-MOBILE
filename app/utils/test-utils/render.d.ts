export type RenderOptions = {
  wrapper?: React.ComponentType<any>
  createNodeMock?: (element: React.ReactElement) => any
}
export type AllTheProvidersProps = { children: React.ReactNode }
