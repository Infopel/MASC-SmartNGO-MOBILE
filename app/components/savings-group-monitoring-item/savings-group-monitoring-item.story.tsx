import * as React from "react"
import { storiesOf } from "@storybook/react-native"
import { StoryScreen, Story, UseCase } from "../../../storybook/views"
import { color } from "../../theme"
import { SavingsGroupMonitoringItem } from "./savings-group-monitoring-item.component"

storiesOf("SavingsGroupMonitoringItem", module)
  .addDecorator((fn) => <StoryScreen>{fn()}</StoryScreen>)
  .add("Style Presets", () => (
    <Story>
      <UseCase text="Primary" usage="The primary.">
        <SavingsGroupMonitoringItem style={{ backgroundColor: color.error }} />
      </UseCase>
    </Story>
  ))
