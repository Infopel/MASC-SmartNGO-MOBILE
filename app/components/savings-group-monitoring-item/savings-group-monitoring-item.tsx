import { Card } from "components/card/card"
import { Text } from "components/text/text"
import { HStack, VStack } from "native-base"
import * as React from "react"
import { StyleProp, ViewStyle } from "react-native"
import { ISavingsGroupMonitoringItem } from "storage/queries/savings-group"
import { DateUtils } from "utils/DateUitls"
import { SavingsGroupMonitoringItemStyles as styles } from "./savings-group-monitoring-item.styles"

export interface SavingsGroupMonitoringItemProps {
  /**
   * An optional style override useful for padding & margin.
   */
  style?: StyleProp<ViewStyle>
  item: ISavingsGroupMonitoringItem
}

/**
 * Describe your component here
 */
export const SavingsGroupMonitoringItem = function SavingsGroupMonitoringItem(props: SavingsGroupMonitoringItemProps) {
  const { style, item } = props
  const containerStyle = {
    ...styles.container,
    ...(style && typeof style === "object" ? style : {}),
  }

  return (
    <Card borderLeftColor={'primary.500'} borderLeftWidth='6' p={2}>
      <HStack space={2}>
        <VStack flex={1} space={2}>
          <Text fontWeight={'bold'}>{item.name}</Text>
          <Text>{DateUtils.prettyDate(item.date)}</Text>
          <Text >Poupan√ßa: {item.savings}</Text>
        </VStack>
        </HStack>
    </Card>
  )
}
