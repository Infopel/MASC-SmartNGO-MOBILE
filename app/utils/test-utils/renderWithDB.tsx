jest.mock("storage/adapter.ts")
import DatabaseProvider from "@nozbe/watermelondb/DatabaseProvider"
import { render } from "@testing-library/react-native"
import { default as React } from "react"
import { database } from "storage/database"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProvidersWithDB({ children }: AllTheProvidersProps) {
    return (
        <DatabaseProvider database={database}>
            <ThemeProvider>{children}</ThemeProvider>
        </DatabaseProvider>
    )
}

export function renderWithDB<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
    return render(ui, { wrapper: AllTheProvidersWithDB, ...options })
}


export class DiagnosticError extends Error {
    name: string = "Diagnostic error"
  }
  