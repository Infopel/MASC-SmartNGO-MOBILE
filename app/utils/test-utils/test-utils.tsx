import { render } from "@testing-library/react-native"
import React from "react"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProviders({ children }: AllTheProvidersProps) {
  return (
    <ThemeProvider>{children}</ThemeProvider>
  )
}

function customRender<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

// re-export everything
export * from "@testing-library/react-native"
// override render method
export { customRender as render }

