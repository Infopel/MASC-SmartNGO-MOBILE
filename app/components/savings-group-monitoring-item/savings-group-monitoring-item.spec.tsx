import { fireEvent, render } from "utils/test-utils/test-utils"
import * as React from "react"
import { SavingsGroupMonitoringItem } from "./savings-group-monitoring-item"



describe("<SavingsGroupMonitoringItem>",()=>{
  it("render", async function () {
    const component = <SavingsGroupMonitoringItem item={{
      age:28,id:"dfsf",name:"Igdrassil",savings:50000
    }}></SavingsGroupMonitoringItem>
  
    expect(render(component).toJSON()).toMatchSnapshot()
  })
})
