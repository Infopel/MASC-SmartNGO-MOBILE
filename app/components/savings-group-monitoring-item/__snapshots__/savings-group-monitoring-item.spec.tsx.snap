// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SavingsGroupMonitoringItem> render 1`] = `
<View
  dataSet={Object {}}
  style={
    Object {
      "backgroundColor": "#fafaf9",
      "borderBottomColor": "#fafafa",
      "borderLeftColor": "#ef8812",
      "borderLeftWidth": 6,
      "borderRadius": 6,
      "borderRightColor": "#fafafa",
      "borderTopColor": "#fafafa",
      "borderWidth": 1,
      "elevation": 2,
      "paddingBottom": 8,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 8,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 1.41,
    }
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      dataSet={Object {}}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        dataSet={Object {}}
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#171717",
            "fontFamily": undefined,
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "700",
            "letterSpacing": 0,
            "lineHeight": 21,
            "textDecorationLine": undefined,
          }
        }
      >
        Igdrassil
      </Text>
      <View
        dataSet={Object {}}
        style={
          Object {
            "height": 8,
          }
        }
      />
      <Text
        dataSet={Object {}}
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#171717",
            "fontFamily": undefined,
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 21,
            "textDecorationLine": undefined,
          }
        }
      >
        28
         anos
      </Text>
      <View
        dataSet={Object {}}
        style={
          Object {
            "height": 8,
          }
        }
      />
      <Text
        dataSet={Object {}}
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#171717",
            "fontFamily": undefined,
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 21,
            "textDecorationLine": undefined,
          }
        }
      >
        Poupan√ßa: 
        50000
      </Text>
    </View>
    <View
      dataSet={Object {}}
      style={
        Object {
          "width": 8,
        }
      }
    />
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      colorScheme="danger"
      dataSet={Object {}}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 4,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
    >
      <Text />
    </View>
  </View>
</View>
`;
